File: api/src/main/java/wcs/java/util/Presentation.java
package wcs.java.util;

import COM.FutureTense.Interfaces.FTValList;
import COM.FutureTense.Interfaces.ICS;

/**
 * Presentation object, helper for Attribute Editors
 * 
 * @author msciab
 * 
 */
public class Presentation {
	private ICS ics;
	private FTValList args = new FTValList();

	public Presentation(ICS ics) {
		this.ics = ics;
	}

	public String get(String presInst, String attribute) {
		String tmp = Util.tmpVar();
		args.removeAll();
		args.setValString("NAME", presInst);
		args.setValString("ATTRIBUTE", attribute);
		args.setValString("VARNAME", tmp);
		ics.runTag("presentation.getprimaryattributevalue", args);
		String result = ics.GetVar(tmp);
		ics.RemoveVar(tmp);
		return result;
	}
}

File: api/todo/Presentation.scala
package wcs.scala
import COM.FutureTense.Interfaces.ICS

object Presentation {
  def apply(pres: String, attr: String)(implicit e: Env) = {
    val v = new wcs.java.util.Presentation(e.ics).get(pres, attr)
    if (v == null || e.isError())
      None
    else Some(v)
  }
}